modules/core_project_factory/locals.tf

```diff
@@ -0,0 +1,32 @@
+/**
+ * Copyright 2018 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+locals {
+  root_path                      = abspath(path.root)
+  preconditions_path             = join("/", [local.root_path, path.module, "scripts", "preconditions"])
+  pip_requirements_absolute_path = join("/", [local.preconditions_path, "requirements.txt"])
+  preconditions_py_absolute_path = join("/", [local.preconditions_path, "preconditions.py"])
+  attributes = {
+    billing_account_attr             = var.billing_account
+    org_id_attr                      = var.org_id
+    credentials_path_attr            = var.credentials_path
+    impersonate_service_account_attr = var.impersonate_service_account
+    folder_id_attr                   = var.folder_id
+    shared_vpc_attr                  = var.shared_vpc
+  }
+  preconditions_command = join(" ", concat([var.python_interpreter_path, local.preconditions_py_absolute_path], [for key, value in local.attributes : ((value != "") ? join(" ", [join("", ["--", key]), local.attributes[key]]) : "")]))

```

Comment:
This is way more complicated than it needs to be. Please separate into different lines and don't use `join` excessively.

Reply:
@morgante I've reworked it. Looks pretty simple in one-line. If It still needs to be separated into different lines, let me know, please.